src/functions/cancellations-processor/index.test.ts

```diff
@@ -0,0 +1,191 @@
+        };
+
+        getDynamoItemSpy.mockResolvedValue(cancellationsSubscription);
+    });
+
+    it.each(["live", "error"] as const)(
+        "correctly processes siri-sx data when the subscription has status of %o",
+        async (status) => {
+            const cancellationsSubscription: CancellationsSubscription = {
+                PK: "411e4495-4a57-4d2f-89d5-cf105441f321",
+                url: "https://mock-data-producer.com/",
+                description: "test-description",
+                shortDescription: "test-short-description",
+                status,
+                requestorRef: null,
+                publisherId: "test-publisher-id",
+                apiKey: "mock-api-key",
+            };
+
+            const valuesMock = vi.fn().mockReturnValue({
+                onConflict: vi.fn().mockReturnValue({
+                    execute: vi.fn().mockResolvedValue(""),
+                }),
+            });
+
+            const dbClient = {
+                insertInto: () => ({
+                    values: valuesMock,
+                }),
+            };
+
+            getDynamoItemSpy.mockResolvedValue(cancellationsSubscription);
+
+            mocks.getS3Object.mockResolvedValueOnce({ Body: { transformToString: () => testSiriSx } });
+            await processSqsRecord(record as S3EventRecord, dbClient as unknown as KyselyDb, "table-name");
+
+            expect(valuesMock).toBeCalledWith(parsedSiriSx);
+        },
+    );
+
+    it("throws an error when the subscription is inactive", async () => {
+        const cancellationsSubscription: CancellationsSubscription = {
+            PK: "411e4495-4a57-4d2f-89d5-cf105441f321",
+            url: "https://mock-data-producer.com/",
+            description: "test-description",
+            shortDescription: "test-short-description",
+            status: "inactive",
+            requestorRef: null,
+            publisherId: "test-publisher-id",
+            apiKey: "mock-api-key",
+        };
+
+        getDynamoItemSpy.mockResolvedValue(cancellationsSubscription);
+
+        await expect(
+            processSqsRecord(record as S3EventRecord, dbClient as unknown as KyselyDb, "table-name"),
+        ).rejects.toThrowError(
+            `Unable to process cancellations for subscription ${mockSubscriptionId} because it is inactive`,
+        );
+
+        expect(valuesMock).not.toHaveBeenCalled();
+
+        expect(putMetricDataSpy).not.toHaveBeenCalledOnce();
+    });
+
+    it("does not insert data into the database when the siri-sx is invalid", async () => {
+        mocks.getS3Object.mockResolvedValueOnce({
+            Body: { transformToString: () => testInvalidSiriSx },
+        });
+
+        await processSqsRecord(record as S3EventRecord, dbClient as unknown as KyselyDb, "table-name");
+
+        expect(valuesMock).not.toHaveBeenCalled();
+    });
+
+    it("does not insert into the database when the siri-sx only contains invalid situations", async () => {
```

Comment:
maybe a test case for a SIRI file with both valid and invalid situations would be good to test that invalid situations get filtered out but valid ones still remain?